src/util/time.h

```diff
@@ -10,12 +10,98 @@
 #include <stdint.h>
 #include <string>
 
+class CClock {
+public:
+    /** Returns the current time in seconds since the POSIX epoch. */
+    virtual int64_t GetTime() const = 0;
+    /** Returns the current time in milliseconds since the POSIX epoch. */
+    virtual int64_t GetTimeMillis() const = 0;
+    /** Returns the current time in microseconds since the POSIX epoch. */
+    virtual int64_t GetTimeMicros() const = 0;
+};
+
+class SystemClock: public CClock {
+private:
+    SystemClock() {}
+    ~SystemClock() {}
+    SystemClock(SystemClock const&)    = delete;
+    SystemClock& operator=(const SystemClock&)= delete;
+public:
+    static SystemClock* Instance() {
+        static SystemClock instance;
+        return &instance;
+    }
+
+    /**
+     * Sets the clock used by zcashd to the system clock.  This must only be
+     * called by `init()`, or in tests.
```

Comment:
Comment needs updating to note the single threaded testing requirement, as discussed in PR gardening.